docs/plugins/plugin-development-guide/implementation.rst

```diff
@@ -0,0 +1,179 @@
+            $this->setAvailableOnDemand(true);
+            $this->isAvailableOnDemand()->shouldReturn(true);
+        }
+    }
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace IronMan\SyliusProductOnDemandPlugin\Entity;
+
+    use Sylius\Component\Core\Model\ProductVariant as BaseProductVariant;
+
+    class ProductVariant extends BaseProductVariant implements ProductVariantInterface
+    {
+        /** @var bool */
+        private $availableOnDemand;
+
+        public function setAvailableOnDemand(?bool $availableOnDemand): void
+        {
+            $this->availableOnDemand = $availableOnDemand;
+        }
+
+        public function isAvailableOnDemand(): ?bool
+        {
+            return $this->availableOnDemand;
+        }
+    }
+
+.. important::
+
+    Of course you need to remember about entity mapping customization as well.
+
+Then our new entity should be configured as sylius product variant resource model:
+
+.. code-block:: yaml
+
+    sylius_product:
+        resources:
+            product_variant:
+                classes:
+                    model: IronMan\SyliusProductOnDemandPlugin\Entity\ProductVariant
+
+This configuration should be placed in ``src/Resources/config/config.yml``. It also has to be imported
+(``- { resource: "@IronManSyliusProductOnDemandPlugin/Resources/config/config.yml" }``) in ``tests/Application/app/config/config.yml``
+to make it work in Behat tests. And at the end importing this file should be one of the steps described in plugin installation.
+
+.. warning::
+
+    Remember that if you modify or add some mapping, you should either provide a migration for the plugin user (that could be
+    copied to their migration folder) or mention in installation process requirement of migration generation!
+
+Form
+****
+
+To make our new field available in Admin panel, form extension is required:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace IronMan\SyliusProductOnDemandPlugin\Form\Extension;
+
+    use Symfony\Component\Form\AbstractTypeExtension;
+    use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+    use Sylius\Bundle\ProductBundle\Form\Type\ProductVariantType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    final class ProductVariantTypeExtension extends AbstractTypeExtension
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options): void
+        {
+            $builder->add('availableOnDemand', CheckboxType::class, [
+                'label' => 'iron_man_sylius_product_on_demand_plugin.ui.available_on_demand',
+            ]);
+        }
+
+        public function getExtendedType(): string
+        {
+            return ProductVariantType::class;
+        }
+    }
+
+Translation keys placed in ``src/Resources/translations/message.{locale}.yml`` will be resolved automatically.
+
+And in your ``app/Resources/config/services.yml``:
+
+.. code-block:: yaml
+
+    services:
```

Comment:
I think these services should be inside of the plugin not the test app. Because you want to load them automatically and not define for each app manually.

Reply:
totally agreed, I was thinking about `src/Resources/config/services.yml`, it's a typo 😄